velox/exec/TaskStats.h

```diff
@@ -67,5 +70,16 @@ struct TaskStats {
   // Epoch time (ms) when the task completed, e.g. all splits were processed and
   // results have been consumed.
   uint64_t endTimeMs{0};
+
+  // Total number of drivers.
+  uint64_t totalDrivers{0};
+  // Completed drivers (that are null in the drivers list).
+  uint64_t completedDrivers{0};
+  // Terminated drivers (return true from isTerminated()).
+  uint64_t terminatedDrivers{0};
+  // 'On thread' drivers.
+  uint64_t runningDrivers{0};
+  // Drivers blocked for various reasons. Based on enum BlockingReason.
+  std::array<uint64_t, 10> blockedDrivers{0, 0, 0, 0, 0, 0, 0, 0, 0};
```

Comment:
I guess this is not perf critical, let's use
```
std::vector<uint64_t> blockedDrivers;
```

Reply:
It is static member based on the enum, no point to use vector.

Reply:
You have to maintain or keep this updated if we add more block reasons which is a little bit fragile?

Reply:
Changing it to [setd](std::map<BlockingReason, uint64_t>)